message_tools/find_invalid_address_case_id_by_uprn.py

```diff
@@ -0,0 +1,73 @@
+import argparse
+import csv
+from pathlib import Path
+
+import requests
+from google.cloud import storage
+from requests import HTTPError
+
+from config import Config
+
+
+def read_invalid_address_uprn_file(file_to_process, project_id):
+    case_id_list = []
+    with open(file_to_process, encoding="utf-8") as open_file_to_process:
+        file_reader = csv.DictReader(open_file_to_process, delimiter=',')
+        print(file_reader)
+
+        for line_number, row in enumerate(file_reader, 1):
+            case_id_list.extend(get_case_id_from_case_api(row['UPRN']))
+
+    address_delta_file = write_invalid_addresses_case_id_file(case_id_list, Path(file_to_process))
+    upload_file_to_bucket(address_delta_file, project_id)
+
+
+def get_case_id_from_case_api(uprn):
+    response = requests.get(f'http://{Config.CASEAPI_HOST}:{Config.CASEAPI_PORT}/cases/uprn/{uprn}')
+    try:
+        response.raise_for_status()
+    except HTTPError:
+        print('Error: invalid or non-existent UPRN')
+        return
```

Comment:
I think this needs more error handling, as is we wouldn't know which lines had failed and this could also mask network problems or 500 errors instead of invalid UPRNs. I think at minimum it needs to log out the error, row number and UPRN in question.